scipy/spatial/distance.py

```diff
@@ -130,12 +132,19 @@ def _copy_array_if_base_present(a):
     return a
 
 
-def _convert_to_bool(X):
-    return np.ascontiguousarray(X, dtype=bool)
+def _correlation_cdist_wrap(XA, XB, dm, **kwargs):
+    XA = XA - XA.mean(axis=1)[:, np.newaxis]
```

Comment:
I prefer `XA.mean(axis=1, keepdims=True)` over adding an axis by indexing, but feel free to leave it as is if you find this clearer.